### DISCUSSION TITLE
Get the last executed query

### DISCUSSION THREAD

**Discussion by nanandn** [TanStack/query#2882] [Q&A]
Is there a way to get the last executed query data? My use case is as follows:
I am displaying a list of items in a data table (ag-grid) component. The data is fetched using react-query, every the filter, sort of pagination conditions change. The grid manages the filter/sort/page models and executes the query. 

I have some action that I can perform on every row/data in the table. When the action (mutation) is executed, I want to optimistically update that row. So I need to get the query result for the query that provided that data, but I don't have the query variables at that point (only the name). How can I use queryClient.getQueryData to get the current list of items shown in the data (basically the result of the last execution of the query). I don't have the complete query key (which includes the filter/sort/page conditions).

Or is there some other means to do what I described?

Thanks
Anand

**Comment by TkDodo** [comment]
you can use `queryClient.getQueriesData` (plural), which supports fuzzy matching on the query key.

**Reply by AlexDiakonov** [reply]
Hi, what if I have dynamic query keys, and i need to take the last one?

**Reply by TkDodo** [reply]
you can use the one with the highest `dataUpdatedAt` value in the `query.state`
